### PR TITLE
feat: add getbalance to nwc provider

### PR DESCRIPTION
**Pull request by im-adithya**
Adds get_balance method in nwc provider

See https://github.com/getAlby/nostr-wallet-connect/issues/113

### CODE REVIEW

File: src/webln/NostrWeblnProvider.ts [hunk 1]
```diff
@@ -175,6 +182,84 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
     }
   }
 
+  // TODO: refactor code in getBalance and sendPayment
```
Comment (rolznz) [inline src/webln/NostrWeblnProvider.ts:185]: @im-adithya can you address the TODO?

Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts:185]: @im-adithya I extracted the method in https://github.com/getAlby/js-sdk/pull/45

If you think it looks good we could use that method in this PR

File: src/webln/NostrWeblnProvider.ts [hunk 2]
```diff
@@ -23,6 +23,13 @@ const NWCs: Record<string,NostrWebLNOptions> = {
   }
 };
 
+// TODO: fetch this from @webbtc/webln-types
```
[the remark this answers is not in the document]
Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts:26]: @reneaaron thanks!

**Comment by bumi** [comment]
I think there are some changes in some `.on syntax in nostr-tools. maybe it's a good time to check that and make sure we use the new syntax

**Comment by rolznz** [comment]
> I think there are some changes in some `.on syntax in nostr-tools. maybe it's a good time to check that and make sure we use the new syntax

@bumi can I create a separate issue and PR for this?

**Review by rolznz — APPROVED** [review]

**Pull request merged**